lambda-assets/device-certificate-generator/app.ts

```diff
@@ -0,0 +1,122 @@
+  InvokeCommand,
+} from '@aws-sdk/client-lambda';
+import {
+  S3Client,
+  GetObjectCommand,
+} from '@aws-sdk/client-s3';
+import {
+  Request,
+  Response,
+} from '@softchef/lambda-events';
+import * as Joi from 'joi';
+import {
+  CertificateGenerator,
+} from '../certificate-generator';
+import {
+  InformationNotFoundError,
+  VerificationError,
+} from '../errors';
+
+export const handler = async (event: any = {}) : Promise <any> => {
+  const request = new Request(event);
+  const response = new Response();
+  const bucketName: string = process.env.BUCKET_NAME!;
+  const bucketPrefix: string = process.env.BUCKET_PREFIX!;
+  const caCertificateId: string = request.input('caCertificateId');
+  const deviceInfo: string = request.input('deviceInfo');
+  try {
+    await verify(caCertificateId, deviceInfo);
+    const caCertificates = await getCaCertificate(caCertificateId, bucketName, bucketPrefix);
+    const deviceCertificates = CertificateGenerator.getDeviceRegistrationCertificates(caCertificates);
+    deviceCertificates.certificate += caCertificates.certificate;
+    return response.json(deviceCertificates);
+  } catch (error) {
+    return response.error(error.stack, error.code);
+  }
+};
+
+async function verify(caCertificateId: string, deviceInfo: string) {
+  const iotClient = new IoTClient({});
+  const { certificateDescription: caCertificateDescription = {} } = await iotClient.send(
+    new DescribeCACertificateCommand({
+      certificateId: caCertificateId,
+    }),
+  );
+  const { certificateArn: caCertificateArn } = await Joi.object({
+    certificateArn: Joi.string().regex(/^arn/).required(),
+  }).unknown(true)
+    .validateAsync(caCertificateDescription).catch((error: Error) => {
+      throw new InformationNotFoundError(error.message);
+    });
+
+  let { tags } = await iotClient.send(
+    new ListTagsForResourceCommand({
+      resourceArn: caCertificateArn,
+    }),
+  );
+  tags = await Joi.array().items(
+    Joi.object({
+      Key: Joi.string().required(),
+      Value: Joi.string(),
+    }).optional(),
+  ).required()
+    .validateAsync(tags).catch((error: Error) => {
+      throw new InformationNotFoundError(error.message);
+    });
+  const { Value: verifierName } = tags!.find(tag => tag.Key === 'verifierName') || { Value: '' };
+
+  if (verifierName) {
+    let {
+      Payload: payload = new Uint8Array(
+        Buffer.from(
+          JSON.stringify({
+            verified: false,
+          }),
+        ),
+      ),
+    } = await new LambdaClient({}).send(
+      new InvokeCommand({
+        FunctionName: decodeURIComponent(verifierName),
+        Payload: Buffer.from(
+          JSON.stringify(deviceInfo),
+        ),
+      }),
+    );
+    const { body } = JSON.parse(
+      String.fromCharCode.apply(null, [...payload]),
+    );
+    await Joi.object({
+      verified: Joi.boolean().allow(true).only().required(),
+    }).required().unknown(true)
+      .validateAsync(body).catch((error: Error) => {
```

Comment:
remove space

Reply:
Do you mean the indentation of line 99?
That is required by eslint: Expected indentation of 6 spaces but found 4.eslint@typescript-eslint/indent

Reply:
#103